lib/hyacinth/ezid/hyacinth_metadata.rb

```diff
@@ -120,14 +120,27 @@ def handle_net_identifier
     def process_names
       return unless @dfd.key? 'name'
       @dfd['name'].each do |name|
-        name['name_role'].each do |role|
-          role_value = role['name_role_term']['value'].downcase
-          case role_value
-          when 'author' then @creators << name['name_term']['value']
-          when 'editor' then @editors << name['name_term']['value']
-          when 'moderator' then @moderators << name['name_term']['value']
-          when 'contributor' then @contributors << name['name_term']['value']
-          end
+        process_name(name)
```

Comment:
@fcd1, I had to create the new method process_name method because rubocop complained about cyclomatic complexity.